### PR TITLE
feat(legend): improve tooltip wording

### LINKED ISSUE #2438: [Legend] Instructions descriptions in tooltip
In order to guide users into the available interactions on top of legends we are going to use the browser tooltip to describe all the possibilities.

This following image shows the expected list of description for each case.

Frame 232

Additional note: hovering a hidden series would display only one line saying "Click to show" since the "⌘ + click" would perform the same action, we will make sure though that even this redundant information will be read by screenreaders.

@dej611 tagging you only for visibility so you don't have to look for this issue.

### PR DESCRIPTION
**Pull request by dej611**
## Summary

The legend title tooltip wording has been improved to be clearer.
Screenshot 2024-05-21 at 15 43 57
Screenshot 2024-05-21 at 15 43 49
Screenshot 2024-05-21 at 15 43 40

## Details

Improved the wording as described in #2438 

## Issue

closes #2438

### CODE REVIEW

**Review by nickofthyme — APPROVED** [review]
Looks fine to me, just mentioned 2 ideas I think would improve the appearance.

cc: @gvnmagni

File: packages/charts/src/components/legend/label.tsx [hunk 1]
```diff
@@ -96,12 +108,12 @@ export function Label({
       tabIndex={0}
       dir={dir}
       className={labelClassNames}
-      title={`${title}${interactionsGuidanceText}`}
+      title={`${title}${getInteractivityTitle(!isSeriesHidden, hiddenSeriesCount, totalSeriesCount)}`}
```
[the remark this answers is not in the document]
Comment (dej611) [reply packages/charts/src/components/legend/label.tsx:111]: WDYT @gvnmagni ?

File: packages/charts/src/components/legend/label.tsx [hunk 2]
```diff
@@ -24,31 +24,45 @@ interface LabelProps {
...
       return `
-Click: ${hideSeriesMessage}
-${modifierKey} + click: ${hideSeriesMessage}`;
+Click ${hideSeriesMessage}`;
     }
     return `
-Click: ${isolateSeriesMessage}
-${modifierKey} + click: ${hideSeriesMessage}`;
+Click ${showSeriesMessage}
+${modifierKey} + click ${hideSeriesMessage}`;
```
Comment (nickofthyme) [inline packages/charts/src/components/legend/label.tsx:45]:
```suggestion
${modifierKey} + Click ${hideSeriesMessage}`;
```

File: packages/charts/src/components/legend/label.tsx [hunk 3]
```diff
@@ -24,31 +24,45 @@ interface LabelProps {
...
-Click: ${showSeriesMessage}
-${modifierKey} + click: ${showSeriesMessage}`;
+Click ${showSeriesMessage}`;
+}
+
+function getInteractivityAriaLabel(isSeriesVisible: boolean, hiddenSeries: number, allSeries: number) {
+  if (isSeriesVisible) {
+    if (allSeries - hiddenSeries === 1) {
+      return `
+Click: ${showAllSeriesMessage}, ${modifierKey} + click: ${hideSeriesMessage}`;
```
Comment (nickofthyme) [inline packages/charts/src/components/legend/label.tsx:55]: These don't matter as much but better to be consistent.

```suggestion
Click: ${showAllSeriesMessage}, ${modifierKey} + Click: ${hideSeriesMessage}`;
```

File: packages/charts/src/components/legend/label.tsx [hunk 4]
```diff
@@ -24,31 +24,45 @@ interface LabelProps {
...
+    }
+    if (hiddenSeries > 0) {
+      return `
+Click: ${hideSeriesMessage}, ${modifierKey} + click: ${hideSeriesMessage}`;
```
Comment (nickofthyme) [inline packages/charts/src/components/legend/label.tsx:59]:
```suggestion
Click: ${hideSeriesMessage}, ${modifierKey} + Click: ${hideSeriesMessage}`;
```

File: packages/charts/src/components/legend/label.tsx [hunk 5]
```diff
@@ -24,31 +24,45 @@ interface LabelProps {
...
+    }
+    return `
+Click: ${showSeriesMessage}, ${modifierKey} + click: ${hideSeriesMessage}`;
```
Comment (nickofthyme) [inline packages/charts/src/components/legend/label.tsx:62]:
```suggestion
Click: ${showSeriesMessage}, ${modifierKey} + Click: ${hideSeriesMessage}`;
```

File: packages/charts/src/components/legend/label.tsx [hunk 6]
```diff
@@ -24,31 +24,45 @@ interface LabelProps {
...
+  }
+  return `
+Click: ${showSeriesMessage}, ${modifierKey} + click: ${showSeriesMessage}`;
```
Comment (nickofthyme) [inline packages/charts/src/components/legend/label.tsx:65]:
```suggestion
Click: ${showSeriesMessage}, ${modifierKey} + Click: ${showSeriesMessage}`;
```

File: packages/charts/src/components/legend/label.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (gvnmagni) [reply packages/charts/src/components/legend/label.tsx:111]: agree! looks great, I know it's a little bigger but I love how easier it is to discern the data series name from the instructions. Thank you @nickofthyme

File: packages/charts/src/components/legend/label.tsx [hunk 7]
```diff
@@ -79,22 +89,20 @@ export function Label({
...
   return isToggleable ? (
     // This div is required to allow multiline text truncation, all ARIA requirements are still met
     // https://stackoverflow.com/questions/68673034/webkit-line-clamp-does-not-apply-to-buttons
     <div
       role="button"
       tabIndex={0}
       dir={dir}
-      className={className}
-      title={`${title}${interactionsGuidanceText}`}
+      className={labelClassNames}
```
Comment (nickofthyme) [inline packages/charts/src/components/legend/label.tsx:99]:
```suggestion
      className={className}
```

**Pull request merged**